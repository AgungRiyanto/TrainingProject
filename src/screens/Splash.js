import React, {useEffect, useState} from 'react';
import {View, Text, StyleSheet, Animated} from 'react-native';

const Splash = ({navigation}) => {
   const [textShow] = useState(new Animated.Value(-20));

   useEffect(() => {
      // timers
      
      Animated.timing(textShow, {
         toValue: 500,
         duration: 2000,
      }).start()

      setTimeout(() => {
         navigation.navigate('Login')
      }, 5000);
   }, []);

   return (
      <View style={styles.container}>
         <Animated.Text
            style={{
               position: 'absolute',
               top: textShow
            }}
         >
            Hello Splash
         </Animated.Text>
      </View>
   )
}

export default Splash;

const styles = StyleSheet.create({
   container: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center'
   }
});