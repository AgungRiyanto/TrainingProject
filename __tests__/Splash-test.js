import React from 'react';
import renderer from 'react-test-renderer';
import Splash from '../src/screens/Splash';

test('renders correctly', () => {
  const tree = renderer.create(<Splash />).toJSON();
  expect(tree).toMatchSnapshot();
});