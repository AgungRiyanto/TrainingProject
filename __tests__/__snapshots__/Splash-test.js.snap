// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    collapsable={false}
    style={
      Object {
        "position": "absolute",
        "top": -20,
      }
    }
  >
    Hello Splash wow
  </Text>
</View>
`;
